In [4]:
%matplotlib inline

import pandas as pd
import matplotlib.pyplot as plt

from IPython.display import Image
from IPython.core.display import HTML 




In [6]:
Image(url= "https://i.cbc.ca/1.3404454.1452808514!/fileImage/httpImage/image.jpg_gen/derivatives/original_620/travelling-jagrs-in-bathroom.jpg")


Komentar o travelling jagrs...

In [10]:
tekme = pd.read_csv('C:\Python34\jagr.csv', index_col = 'st_tekme')

In [3]:
tekme

starost nasprotnik  zadetki  asistence  tocke  plus_minus  kazni  \
st_tekme                                                                    
1         18-232        WSH        0          0      0           2      0   
2         18-234        NJD        1          0      1          -1      0   
3         18-236        STL        0          0      0          -1      0   
4         18-238        CHI        0          0      0           0      0   
5         18-240        NYI        1          0      1           1      0   
6         18-243        PHI        0          0      0          -2      0   
7         18-246        BUF        1          1      2           1      2   
8         18-247        NYR        0          0      0          -2      2   
9         18-250        MTL        0          0      0           0      0   
10        18-252        QUE        0          0      0           0      0   
11        18-254        NJD        0          0      0          -2      0   
12        18-255        NYI        2          0      2           2      0   
13        18-257        PHI        0          2      2           1      0   
14        18-261        NYR        0          0      0           0      0   
15        18-264        CGY        0          0      0           1      2   
16        18-266        STL        0          0      0           0      2   
17        18-268        BOS        0          1      1           0      0   
18        18-271        MNS        1          0      1           0      0   
19        18-272        WIN        1          0      1          -1      0   
20        18-275        LAK        0          0      0           0      0   
21        18-279        PHI        0          0      0           0      0   
22        18-281        WSH        0          0      0          -2      0   
23        18-282        WSH        0          0      0          -1      0   
24        18-285        EDM        0          0      0          -1      0   
25        18-287        HAR        0          0      0          -1      2   
26        18-289        MNS        0          0      0           0      0   
27        18-291        NYR        0          1      1           0      0   
28        18-293        WSH        0          0      0           0      0   
29        18-295        VAN        0          0      0           0      0   
30        18-296        HAR        0          0      0          -1      2   
...          ...        ...      ...        ...    ...         ...    ...   
1682      44-360        LAK        0          1      1          -1      2   
1683      44-362        NSH        1          0      1           2      0   
1684      45-000        SJS        0          1      1           2      2   
1685      45-002        ANA        1          0      1           1      0   
1686      45-003        LAK        0          1      1           2      0   
1687      45-005        STL        0          0      0           0      0   
1688      45-007        EDM        0          0      0           0      0   
1689      45-009        CGY        0          0      0          -2      2   
1690      45-011        OTT        0          0      0          -1      0   
1691      45-013        CAR        1          0      1           0      0   
1692      45-015        PHI        0          0      0           0      2   
1693      45-017        DAL        0          0      0           0      0   
1694      45-020        NYR        0          0      0           0      0   
1695      45-023        MIN        1          0      1           0      0   
1696      45-024        TBL        1          0      1           2      0   
1697      45-027        TOR        0          0      0           1      2   
1698      45-029        CBJ        0          0      0           0      0   
1699      45-030        NYR        0          2      2           2      0   
1700      45-032        PIT        0          0      0          -2      0   
1701 

In [29]:
tekme[['zadetki','asistence','tocke']].sum()

zadetki       765
asistence    1149
tocke        1914
dtype: int64

# Individualni dosežki

Jaromir Jagr je v svoji dolgoletni karieri dosegel 765 zadetkov, 1149 podaj za skupnih 1914 točk, kar ga uvršča na drugo mesto najučinkovitejših hokejistov lige NHL vseh časov. Za začetek si bomo ogledali najboljše individualne dosežke Jaromirja na posameznih tekmah.

In [26]:
tekme[['zadetki','asistence','tocke','plus_minus','kazni','streli']].max()

zadetki        4
asistence      5
tocke          7
plus_minus     5
kazni         17
streli        12
dtype: int64

In [40]:
tekme[tekme['zadetki']>2][['nasprotnik','zadetki']]

nasprotnik  zadetki
st_tekme                    
54              BOS        3
323             BOS        3
453             OTT        3
464             BOS        3
681             OTT        3
697             NYI        3
729             NYR        4
757             MTL        3
793             CGY        3
917             FLA        3
928             TBL        3
1036            NYI        3
1046            PIT        3
1096            PHI        3
1513            PHI        3

V svoji karieri je dosegel kar 15 tako imenovanih "Hat-trickov", enkrat pa mu je uspelo zadeti kar štirikrat. To se je zgodilo na njegovi 729. tekmi proti ekipi New York Rangers.

In [38]:
tekme[tekme['asistence']>3][['nasprotnik','asistence']]

nasprotnik  asistence
st_tekme                      
251             PHI          5
324             HAR          4
352             NYR          4
423             WIN          4
474             STL          4
627             BUF          4
697             NYI          4
917             FLA          4
1100            NYI          4
1253            MTL          4

Največ asistenc na eni tekmi je dosegel proti ekipi Philadelphia Flyers in sicer 5. 

In [50]:
tekme[tekme['tocke']>5][['nasprotnik','zadetki','asistence','tocke']]

nasprotnik  zadetki  asistence  tocke
st_tekme                                      
251             PHI        1          5      6
697             NYI        3          4      7
917             FLA        3          4      7

Dvakrat v karieri mu je na eni tekmi uspelo doseči kar 7 točk! Pri tem je obakrat dosegel po 3 zadetke in 4 asistence.

In [51]:
tekme[tekme['kazni']>9][['nasprotnik','kazni']]

nasprotnik  kazni
st_tekme                  
35              WIN     10
128             WSH     12
212             PHI     15
248             STL     15
344             NYI     17
467             WSH     12
707             ATL     10

Le šestkrat se mu je na tekmah lige NHL zgodilo, da je prejel vsaj 10 kazenskih minut. Največ, 17, na tekmi proti ekipi New York Islanders. 

In [36]:
tekme[(tekme['plus_minus']<-4) | (tekme['plus_minus']>4)][['nasprotnik','plus_minus']]

nasprotnik  plus_minus
st_tekme                       
98              EDM           5
104             PHI           5
690             CAR           5
817             OTT          -5
917             FLA           5

Plus_minus (+/-) je statistika, ki pove pri koliko danih oziroma prejetih golih je bil hokejist na ledu. Za vsak dan zadetek ob katerem je bil hokejist takrat ne ledu, se poveča za 1 in podobno, za vsak prejet zmanjša za 1. Jaromir Jagr je v svoji karieri štirikrat zabeležil najboljši  +/- 5, svojega najslabšega pa enkrat, to je -5.